server/app/services/cloud/azure/AzurePublicStorage.java

```diff
@@ -31,4 +76,30 @@ protected String getPublicDisplayUrlInternal(String fileKey) {
   public void prunePublicFileStorage(ImmutableSet<String> inUseFileKeys) {
     throw new UnsupportedOperationException("not implemented");
   }
+
+  interface Client {
+
+    String getSasToken(String fileName, Optional<String> originalFileName);
+
+    String getBlobUrl(String fileName);
+  }
+
+  /** Class to use for BlobStorage unit tests. */
+  static class NullClient implements Client {
+
+    NullClient() {}
+
+    @Override
+    public String getSasToken(String fileName, Optional<String> originalFileName) {
+      if (originalFileName.isPresent()) {
+        return "sasTokenWithContentHeaders";
+      }
+      return "sasToken";
+    }
+
+    @Override
+    public String getBlobUrl(String fileName) {
+      return "http://www.blobUrl.com";
```

Comment:
(blocking) Set this to localhost. We don't want to possible send data to a domain we don't control.

Reply:
done